sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/RebaseDateTime.scala

```diff
@@ -186,7 +186,7 @@ object RebaseDateTime {
       .setTimeZone(TimeZoneUTC)
       .setDate(localDate.getYear, localDate.getMonthValue - 1, localDate.getDayOfMonth)
       .build()
-    Math.toIntExact(Math.floorDiv(utcCal.getTimeInMillis, MILLIS_PER_DAY))
+    Math.toIntExact(utcCal.getTimeInMillis, MILLIS_PER_DAY)
```

Comment:
Where is the division?

Reply:
Yes, the comma needs to become a slash. 
I think this change would then be OK as the sign of these two args are always the same.
However, does it make much difference? it's going to be a few extra math ops. It might be useful to just keep this consistent

Reply:
Oops Sorry, silly me. 
And yes @srowen, this does make things faster. I can do some benchmarking if you would like me to...

Reply:
I guess I'd be surprised if the operations were meaningfully faster, given how much else is happening. I'm sure a microbenchmark of a trillion calls would show a difference, but that's not quite the use case.
Nevertheless are there other instances where both args are definitely nonnegative?

Reply:
Here, the question is not about performance but about our understanding of the logic.
If we set only date fields:
```scala
.setDate(localDate.getYear, localDate.getMonthValue - 1, localDate.getDayOfMonth)
```
the time part MUST be zero. If it is not, something is wrong going on.

I would add the assert before the division:
```scala
assert(utcCal.getTimeInMillis % MILLIS_PER_DAY == 0)
```